CMakeLists.txt

```diff
@@ -115,6 +115,10 @@ foreach(TARGET sonata_shared sonata_static)
     target_compile_options(${TARGET}
         PRIVATE ${SONATA_COMPILE_OPTIONS}
     )
+    target_link_libraries(${TARGET}
+        PRIVATE fmt::fmt-header-only
+        PUBLIC HighFive
```

Comment:
Why did this go from PRIVATE to PUBLIC?

Reply:
Because of: https://github.com/BlueBrain/libsonata/blob/master/include/bbp/sonata/report_reader.h#L9

Reply:
Could that not be forward-declared? :(

Reply:
not easily, can we do this in a new PR?
I can create a ticket for that

Reply:
I think we should do that first, so that HighFive doesn't bleed through the public headers; then we don't need to make it `PUBLIC`, which should be avoided, imo.

Reply:
ok :(

Reply:
This is not an easy patch, and today highfive is already a public dependency, but hidden (people have to guess it ^^)

Reply:
Good point.

However, isn't it a matter of wrapping `ReportReader::file_` in `std::unique_ptr`, and saving the population name instead of `pop_group_`?

Reply:
Good idea, i was trying to put `pop_group_` inside `std::unique_ptr` too, it was leading to complicate things ^^

Reply:
It implies to store the file in the population class... That is the same, do we really want a shared_ptr?